pros/src/adi/analog.rs

```diff
@@ -0,0 +1,116 @@
+use crate::adi::{
+    AdiError,
+    AdiSlot
+};
+
+use crate::error::bail_on;
+
+use pros_sys::PROS_ERR;
+
+pub struct AdiAnalogIn {
+    port: u8,
+}
+
+impl AdiAnalogIn {
+    /// Create an AdiAnalogIn without checking if it is valid.
+    /// 
+    /// # Safety
+    /// 
+    /// The port must be above 0 and below [`pros_sys::NUM_ADI_PORTS`].
+    pub unsafe fn new_unchecked(slot: AdiSlot) -> Self {
+        Self {
+            port: slot as u8
+        }
+    }
+    
+    /// Create an AdiAnalogIn, throwing an error if the port is invalid.
+    pub fn new(slot: AdiSlot) -> Self {
+        let port = slot as u8;
+        if port < 1 || port > {pros_sys::NUM_ADI_PORTS as u8} {
+            panic!("Invalid ADI port");
+        }
+        Self { port }
+    }
+
+    /// Create an AdiAnalogIn, returning err `AdiError::InvalidPort` if the port is invalid.
+    pub fn try_new(slot: AdiSlot) -> Result<Self, AdiError> {
```

Comment:
This isn't necessary, but it might align more with the rest of the pros-rs API if new returned a ``Result<Self, AdiError>`` and there was a differently named method for the current new function.